### PR TITLE
feature/add check department logic

### PR DESCRIPTION
**Pull request by borovHokage**

### CODE REVIEW

File: src/NewsService.Business/CreateNewsCommand.cs [hunk 1]
```diff
@@ -17,32 +25,77 @@ public class CreateNewsCommand : ICreateNewsCommand
...
             }
 
+            OperationResultResponse<Guid> response = new();
+
             _validator.ValidateAndThrowCustom(request);
 
-            var news = _mapper.Map(request);
+            List<Guid> existDepartments = CheckDepartmentExistence(request.DepartmentId, response.Errors);
+
+            var news = _mapper.Map(request, existDepartments);
```
Comment (MariyaSin) [inline src/NewsService.Business/CreateNewsCommand.cs:91]: do not use var

File: src/NewsService.Business/CreateNewsCommand.cs [hunk 2]
```diff
@@ -17,32 +25,77 @@ public class CreateNewsCommand : ICreateNewsCommand
...
 
-            return _repository.CreateNews(news);
+            return new OperationResultResponse<Guid>
+            {
+                Body = _repository.CreateNews(news),
+                Status = OperationResultStatusType.FullSuccess,
+                Errors = response.Errors
```
Comment (MariyaSin) [inline src/NewsService.Business/CreateNewsCommand.cs:97]: wtf???))))

File: src/NewsService.Business/Interfaces/ICreateNewsCommand.cs [hunk 3]
```diff
@@ -16,6 +17,8 @@ public interface ICreateNewsCommand
         /// </summary>
         /// <param name="request">News data.</param>
         /// <returns>Guid of added news.</returns>
-        Guid Execute(News request);
+        
+        OperationResultResponse<Guid> Execute(News request);
+
```
Comment (MariyaSin) [inline src/NewsService.Business/Interfaces/ICreateNewsCommand.cs:22]: remove empty line

File: src/NewsService.Validation/NewsValidator.cs [hunk 4]
```diff
@@ -22,6 +23,13 @@ public NewsValidator()
 
             RuleFor(news => news.Content)
                 .NotEmpty();
+
+            When(
+                news => news.DepartmentId.HasValue,
+                () =>
+                    RuleFor(news => news.DepartmentId)
+                        .Must(DepartmentId => DepartmentId != Guid.Empty)
```
Comment (MariyaSin) [inline src/NewsService.Validation/NewsValidator.cs:31]: .NotEmpty()
It is does not work?

File: src/NewsService.Business/CreateNewsCommand.cs [hunk 5]
```diff
@@ -17,32 +25,77 @@ public class CreateNewsCommand : ICreateNewsCommand
         private readonly IDbNewsMapper _mapper;
         private readonly INewsValidator _validator;
         private readonly IAccessValidator _accessValidator;
+        private readonly IRequestClient<ICheckDepartmentsExistence> _rcCheckDepartmentsExistence;
+        private readonly ILogger<CreateNewsCommand> _logger;
+
+        private List<Guid> CheckDepartmentExistence(Guid? departmentId, List<string> errors)
+        {
+            string errorMessage = "Failed to check the existing department.";
+            string logMessage = "Department not found. {departmentIds}";
```
Comment (BelehovEgor) [inline src/NewsService.Business/CreateNewsCommand.cs:34]: mb Department with id {id} not found.

File: src/NewsService.Business/CreateNewsCommand.cs [hunk 6]
```diff
@@ -17,32 +25,77 @@ public class CreateNewsCommand : ICreateNewsCommand
...
+        public OperationResultResponse<Guid> Execute(News request)
         {
             if (!(_accessValidator.IsAdmin() ||
-                  _accessValidator.HasRights(Rights.AddEditRemoveNews)))
+                 _accessValidator.HasRights(Rights.AddEditRemoveNews)))
             {
                 throw new ForbiddenException("Not enough rights.");
             }
 
+            OperationResultResponse<Guid> response = new();
```
Comment (BelehovEgor) [inline src/NewsService.Business/CreateNewsCommand.cs:85]: create after validate

**Review by BelehovEgor — APPROVED** [review]

File: src/NewsService.Business/CreateNewsCommand.cs (hunk 6 above)
Comment (borovHokage) [reply src/NewsService.Business/CreateNewsCommand.cs:85]: fixed

File: src/NewsService.Business/CreateNewsCommand.cs (hunk 2 above)
Comment (borovHokage) [reply src/NewsService.Business/CreateNewsCommand.cs:97]: fixed

File: src/NewsService.Business/CreateNewsCommand.cs (hunk 1 above)
Comment (borovHokage) [reply src/NewsService.Business/CreateNewsCommand.cs:91]: fixed

File: src/NewsService.Business/Interfaces/ICreateNewsCommand.cs (hunk 3 above)
Comment (borovHokage) [reply src/NewsService.Business/Interfaces/ICreateNewsCommand.cs:22]: fixed

File: src/NewsService.Validation/NewsValidator.cs (hunk 4 above)
Comment (borovHokage) [reply src/NewsService.Validation/NewsValidator.cs:31]: yes

File: src/NewsService.Business/CreateNewsCommand.cs (hunk 5 above)
Comment (borovHokage) [reply src/NewsService.Business/CreateNewsCommand.cs:34]: done

File: src/NewsService.Business/CreateNewsCommand.cs [hunk 7]
```diff
@@ -17,32 +25,74 @@ public class CreateNewsCommand : ICreateNewsCommand
...
+                if (response.Message.IsSuccess)
+                {
+                    if (!response.Message.Body.DepartmentIds.Any())
+                    {
+                        errors.Add($"Department Id: {departmentId} does not exist");
+                    }
+                    return response.Message.Body.DepartmentIds;
+                }
+
+                _logger.LogWarning($"Can not find {departmentId} with this Ids '{departmentId}': " +
```
Comment (djino97) [inline src/NewsService.Business/CreateNewsCommand.cs:49]: use parameters logger instead use interpolation

File: src/NewsService.Business/CreateNewsCommand.cs [hunk 8]
```diff
@@ -17,32 +25,74 @@ public class CreateNewsCommand : ICreateNewsCommand
...
+                    $"{Environment.NewLine}{string.Join('\n', response.Message.Errors)}");
+            }
+            catch (Exception exc)
+            {
+                _logger.LogError(exc, logMessage);
+            }
+
+            errors.Add(errorMessage);
+            return new List<Guid>();
```
Comment (djino97) [inline src/NewsService.Business/CreateNewsCommand.cs:58]: need return null, so as Body response should be null value in this case

File: src/NewsService.Validation/NewsValidator.cs (hunk 4 above)
Comment (djino97) [reply src/NewsService.Validation/NewsValidator.cs:31]: Why?

File: src/NewsService.Business/CreateNewsCommand.cs (hunk 7 above)
Comment (borovHokage) [reply src/NewsService.Business/CreateNewsCommand.cs:49]: Done

File: src/NewsService.Business/CreateNewsCommand.cs [hunk 9]
```diff
@@ -17,32 +25,79 @@ public class CreateNewsCommand : ICreateNewsCommand
...
+                {
+                    if (!response.Message.Body.DepartmentIds.Any())
+                    {
+                        errors.Add($"Department Id: {departmentId} does not exist");
+                        return null;
+                    }
+                    return response.Message.Body.DepartmentIds;
+                }
+
+                _logger.LogWarning("Can not find {departmentId} with this Id: {departmentId}: " +
```
Comment (MariyaSin) [inline src/NewsService.Business/CreateNewsCommand.cs:54]: Can not find department with Id: {departmentId}:

File: src/NewsService.Business/CreateNewsCommand.cs [hunk 10]
```diff
@@ -17,32 +25,79 @@ public class CreateNewsCommand : ICreateNewsCommand
...
 
             _validator.ValidateAndThrowCustom(request);
 
-            var news = _mapper.Map(request);
+            OperationResultResponse<Guid> response = new();
+
+            List<Guid> existDepartments = CheckDepartmentExistence(request.DepartmentId, response.Errors);
+
+            response.Body = _repository.CreateNews(_mapper.Map(request, existDepartments));
+
```
Comment (MariyaSin) [inline src/NewsService.Business/CreateNewsCommand.cs:97]: remove empty line

File: src/NewsService.Business/CreateNewsCommand.cs (hunk 9 above)
Comment (borovHokage) [reply src/NewsService.Business/CreateNewsCommand.cs:54]: fixed

**Review by MariyaSin — APPROVED** [review]

File: src/NewsService.Business/CreateNewsCommand.cs (hunk 10 above)
Comment (borovHokage) [reply src/NewsService.Business/CreateNewsCommand.cs:97]: fixed

**Review by Cheephot — APPROVED** [review]

File: src/NewsService.Business/CreateNewsCommand.cs (hunk 8 above)
Comment (borovHokage) [reply src/NewsService.Business/CreateNewsCommand.cs]: fixed

**Review by VLiamin — APPROVED** [review]

**Pull request merged**